Sai Moola- Create choosing village from the map backend

saishekhar10 opened the pull request:
# Description
<img width="698" alt="Screenshot 2025-03-20 at 2 40 47 PM" src="https://github.com/user-attachments/assets/3eddce4f-717c-4373-ab47-f9a7bf11c35b" />

## Related PRS (if any):
None
…

## Main changes explained:
Created a schema for the Villages and a controller file that contains endpoints for creating, modifying and fetching information about villages.
…

## How to test:
1. check into current branch
2. do `npm install` and `...` to run this PR locally
3. Login as admin user using this link on postman and get your login token http://localhost:4500/api/login ( use a POST request )
4. Create a GET request on postman using this link: http://localhost:4500/api/villages, run this request using the login token that you got as your Authorization header. You should be able to see a list of 8 Villages. 
<img width="1347" alt="Screenshot 2025-03-20 at 2 48 51 PM" src="https://github.com/user-attachments/assets/a35e6563-d325-4088-8a45-69cfe784055f" />

5.Create a GET request on postman using this link: http://localhost:4500/api/villages/region/C ( don't forget to include your authorization token ), you should be able to see information about the village with Region ID: C 
<img width="1348" alt="Screenshot 2025-03-20 at 2 51 04 PM" src="https://github.com/user-attachments/assets/048fd4a8-1cd8-4247-aaa7-4349bad3f876" />

Similarly, test the same endpoint to check if you're able to get information about the rest of the villages using their respective Region Id's
Use these GET requests to do so: 
http://localhost:4500/api/villages/region/1
http://localhost:4500/api/villages/region/2
http://localhost:4500/api/villages/region/2
http://localhost:4500/api/villages/region/3
http://localhost:4500/api/villages/region/4
http://localhost:4500/api/villages/region/5
http://localhost:4500/api/villages/region/6
http://localhost:4500/api/villages/region/7


## Screenshots or videos of changes:

## Note:
Please don't test the endpoints for creating or editing villages as of now.

fangedShadow commented:
can you change your folder name from lb_dashboard to lbdashboard as shown in the ss. As we other working in the feature subsets are naming it that and so that there won't be different names for the that when they are merged
![image_2025-03-20_184845395](https://github.com/user-attachments/assets/628b081e-44e7-424d-9bc2-f79b15a9d1ec)

fangedShadow commented:
> > can you change your folder name from lb_dashboard to lbdashboard as shown in the ss. As we other working in the feature subsets are naming it that and so that there won't be different names for the that when they are merged ![image_2025-03-20_184845395](https://private-user-images.githubusercontent.com/75034643/425210110-628b081e-44e7-424d-9bc2-f79b15a9d1ec.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDI1MTEzMjIsIm5iZiI6MTc0MjUxMTAyMiwicGF0aCI6Ii83NTAzNDY0My80MjUyMTAxMTAtNjI4YjA4MWUtNDRlNy00MjRkLTliYzItZjc5YjE1YTlkMWVjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAzMjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMzIwVDIyNTAyMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTg4NjQ0NzAwZmRkNGEyZDUyY2RiNTJkYjZhNDI3ODE3MTA5MzM4Y2RlMzVhOWYwN2FjYjY2NjY4MzYxYzRiNjgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.zDFtzTD68Diaeb3yXjyb5o0TdkG7v5GWV6au8B7eYTc)

great, 
I just realized that I was working on the frontend of the map. I set the positioning this way so that the map pointers (markers) work properly. The markers need to be placed within the correct positioned container so that they align accurately on the map. I attached a ss so that it shows how data is used in the frontend for it. 
![Screenshot 2025-03-20 185124](https://github.com/user-attachments/assets/ca9f286e-5c76-4f26-8f58-2856e4276b77)

saishekhar10 commented:
> can you change your folder name from lb_dashboard to lbdashboard as shown in the ss. As we other working in the feature subsets are naming it that and so that there won't be different names for the that when they are merged ![image_2025-03-20_184845395](https://private-user-images.githubusercontent.com/75034643/425210110-628b081e-44e7-424d-9bc2-f79b15a9d1ec.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDI1MTE2MTIsIm5iZiI6MTc0MjUxMTMxMiwicGF0aCI6Ii83NTAzNDY0My80MjUyMTAxMTAtNjI4YjA4MWUtNDRlNy00MjRkLTliYzItZjc5YjE1YTlkMWVjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAzMjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMzIwVDIyNTUxMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWRjMmUxYzEyOTFiY2U0ZDkyZTVmZTFkNzM0Y2ZlMzMwOGMzNzA4NjhiODExNmFhYzZhZDAyZDZhOTJmYzdlMDAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.u7fJVghMi1ZjXLscOClJoaND37sMGMz1bhGlb-UZK68)

I've changed the folder names to "lbdashboard"

saishekhar10 commented:
> > > can you change your folder name from lb_dashboard to lbdashboard as shown in the ss. As we other working in the feature subsets are naming it that and so that there won't be different names for the that when they are merged ![image_2025-03-20_184845395](https://private-user-images.githubusercontent.com/75034643/425210110-628b081e-44e7-424d-9bc2-f79b15a9d1ec.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDI1MTEzMjIsIm5iZiI6MTc0MjUxMTAyMiwicGF0aCI6Ii83NTAzNDY0My80MjUyMTAxMTAtNjI4YjA4MWUtNDRlNy00MjRkLTliYzItZjc5YjE1YTlkMWVjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAzMjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMzIwVDIyNTAyMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTg4NjQ0NzAwZmRkNGEyZDUyY2RiNTJkYjZhNDI3ODE3MTA5MzM4Y2RlMzVhOWYwN2FjYjY2NjY4MzYxYzRiNjgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.zDFtzTD68Diaeb3yXjyb5o0TdkG7v5GWV6au8B7eYTc)
> 
> great, I just realized that I was working on the frontend of the map. I set the positioning this way so that the map pointers (markers) work properly. The markers need to be placed within the correct positioned container so that they align accurately on the map. I attached a ss so that it shows how data is used in the frontend for it. Can you include this in the model as well. ![Screenshot 2025-03-20 185124](https://private-user-images.githubusercontent.com/75034643/425211309-ca9f286e-5c76-4f26-8f58-2856e4276b77.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDI1MTM2ODQsIm5iZiI6MTc0MjUxMzM4NCwicGF0aCI6Ii83NTAzNDY0My80MjUyMTEzMDktY2E5ZjI4NmUtNWM3Ni00ZjI2LThmNTgtMjg1NmU0Mjc2Yjc3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAzMjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMzIwVDIzMjk0NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTBlMTM0ODY3MGIzYjU2YWJkOGYyMWVkNTMwMjI2ZjY5MzY3MDY4NzU3NjYzYmM1YmQwYWQ1ZTM4MTk2MWMyOTMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.Ux6p7p_YtN65qaK_zeXORTAwd3tCqJHR4eCZXoPUx4U)

I already have a field in my schema called mapCoordinates which was intended to store positions of each village in the map. Would you like me to replace "mapCoordinates" to "positions"?

fangedShadow commented:
> > > > can you change your folder name from lb_dashboard to lbdashboard as shown in the ss. As we other working in the feature subsets are naming it that and so that there won't be different names for the that when they are merged ![image_2025-03-20_184845395](https://private-user-images.githubusercontent.com/75034643/425210110-628b081e-44e7-424d-9bc2-f79b15a9d1ec.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDI1MTEzMjIsIm5iZiI6MTc0MjUxMTAyMiwicGF0aCI6Ii83NTAzNDY0My80MjUyMTAxMTAtNjI4YjA4MWUtNDRlNy00MjRkLTliYzItZjc5YjE1YTlkMWVjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAzMjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMzIwVDIyNTAyMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTg4NjQ0NzAwZmRkNGEyZDUyY2RiNTJkYjZhNDI3ODE3MTA5MzM4Y2RlMzVhOWYwN2FjYjY2NjY4MzYxYzRiNjgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.zDFtzTD68Diaeb3yXjyb5o0TdkG7v5GWV6au8B7eYTc)
> > 
> > 
> > great, I just realized that I was working on the frontend of the map. I set the positioning this way so that the map pointers (markers) work properly. The markers need to be placed within the correct positioned container so that they align accurately on the map. I attached a ss so that it shows how data is used in the frontend for it. Can you include this in the model as well. ![Screenshot 2025-03-20 185124](https://private-user-images.githubusercontent.com/75034643/425211309-ca9f286e-5c76-4f26-8f58-2856e4276b77.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDI1MTM2ODQsIm5iZiI6MTc0MjUxMzM4NCwicGF0aCI6Ii83NTAzNDY0My80MjUyMTEzMDktY2E5ZjI4NmUtNWM3Ni00ZjI2LThmNTgtMjg1NmU0Mjc2Yjc3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAzMjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMzIwVDIzMjk0NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTBlMTM0ODY3MGIzYjU2YWJkOGYyMWVkNTMwMjI2ZjY5MzY3MDY4NzU3NjYzYmM1YmQwYWQ1ZTM4MTk2MWMyOTMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.Ux6p7p_YtN65qaK_zeXORTAwd3tCqJHR4eCZXoPUx4U)
> 
> I already have a field in my schema called mapCoordinates which was intended to store positions of each village in the map. Would you like me to replace "mapCoordinates" to "positions"?

yes, because map coordinates would sugest that it is a world cooridnates rather than position on the image which we current use. also if you could have it be like this: 
position : {
left: string
right: string
} and have it in the database as well as, also description text rather than description link because we have to provide some information when they pinpoint the village.

saishekhar10 commented:
> > > > > can you change your folder name from lb_dashboard to lbdashboard as shown in the ss. As we other working in the feature subsets are naming it that and so that there won't be different names for the that when they are merged ![image_2025-03-20_184845395](https://private-user-images.githubusercontent.com/75034643/425210110-628b081e-44e7-424d-9bc2-f79b15a9d1ec.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDI1MTEzMjIsIm5iZiI6MTc0MjUxMTAyMiwicGF0aCI6Ii83NTAzNDY0My80MjUyMTAxMTAtNjI4YjA4MWUtNDRlNy00MjRkLTliYzItZjc5YjE1YTlkMWVjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAzMjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMzIwVDIyNTAyMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTg4NjQ0NzAwZmRkNGEyZDUyY2RiNTJkYjZhNDI3ODE3MTA5MzM4Y2RlMzVhOWYwN2FjYjY2NjY4MzYxYzRiNjgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.zDFtzTD68Diaeb3yXjyb5o0TdkG7v5GWV6au8B7eYTc)
> > > 
> > > 
> > > great, I just realized that I was working on the frontend of the map. I set the positioning this way so that the map pointers (markers) work properly. The markers need to be placed within the correct positioned container so that they align accurately on the map. I attached a ss so that it shows how data is used in the frontend for it. Can you include this in the model as well. ![Screenshot 2025-03-20 185124](https://private-user-images.githubusercontent.com/75034643/425211309-ca9f286e-5c76-4f26-8f58-2856e4276b77.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDI1MTM2ODQsIm5iZiI6MTc0MjUxMzM4NCwicGF0aCI6Ii83NTAzNDY0My80MjUyMTEzMDktY2E5ZjI4NmUtNWM3Ni00ZjI2LThmNTgtMjg1NmU0Mjc2Yjc3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAzMjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMzIwVDIzMjk0NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTBlMTM0ODY3MGIzYjU2YWJkOGYyMWVkNTMwMjI2ZjY5MzY3MDY4NzU3NjYzYmM1YmQwYWQ1ZTM4MTk2MWMyOTMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.Ux6p7p_YtN65qaK_zeXORTAwd3tCqJHR4eCZXoPUx4U)
> > 
> > 
> > I already have a field in my schema called mapCoordinates which was intended to store positions of each village in the map. Would you like me to replace "mapCoordinates" to "positions"?
> 
> yes, because map coordinates would sugest that it is a world cooridnates rather than position on the image which we current use. also if you could have it be like this: position : { left: string right: string } and have it in the database as well as when we connect it it won't have to changed on the backend (position that i used are [here](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/3293/files#diff-9d05ba17ed9e8a08e7831b536898a12b6c3797dee4afccbc889f6ad7ad92d804)) under masterplan.jsx, also description text rather than description link because we have to provide some information when they pinpoint the village.

I've updated the schema and also updated the documents in the database to include the descriptions and position values as you have it in your jsx file.

sidhu47h reviewed (approved):
Tested the api endpoints using postman and got expected responses.
![Screenshot 2025-03-21 at 3 45 33 PM](https://github.com/user-attachments/assets/311eadf2-0461-4a07-9fac-ad570930360e)
![Screenshot 2025-03-21 at 3 47 22 PM](https://github.com/user-attachments/assets/7e495731-c96f-4fdf-b4ab-e4b9af699678)

nikhilrouth reviewed (approved):
Tested the PR  and everything is working as expected and also tested with non-existing Region IDs, and error handling works correctly so **approving this PR 1280**.

<img width="1053" alt="Screenshot 2025-03-21 at 3 30 31 PM" src="https://github.com/user-attachments/assets/790f632a-f088-4842-a5ee-ca481f689cbf" />

<img width="1055" alt="Screenshot 2025-03-21 at 3 31 44 PM" src="https://github.com/user-attachments/assets/ecae58c1-6675-451c-bbe8-9dbdb22659eb" />

manushajyasta30 commented:
<img width="661" alt="image" src="https://github.com/user-attachments/assets/0612011d-85ba-4918-963c-9df2d856c481" />

Amalesh-A commented:
I checked out your branch and followed your steps. I was able to run and successfully with expected outputs.

![1](https://github.com/user-attachments/assets/f65b0ea3-c990-4e75-98d8-cb2e32ccd105)
![2](https://github.com/user-attachments/assets/a638b096-6280-468a-bda9-5889ee9f04f3)
![3](https://github.com/user-attachments/assets/c30ad4f5-b163-45cd-bd9a-ca0ea45fbe39)
![4](https://github.com/user-attachments/assets/b3eb1264-49a6-47dd-b1d0-df5c78786af1)

humemah commented:
I tested the API endpoints locally as per the given instructions. This included logging in as an admin, retrieving a token, and making GET requests to fetch village data by region.
 /api/villages returned 8 villages.
/api/villages/region/{id} returned correct data for all regions.No issues found.


![486567875_588275934086417_8731695356143243492_n](https://github.com/user-attachments/assets/b6e44dae-492a-45a7-93f1-8906346a2825)
![PR 1287](https://github.com/user-attachments/assets/db92df07-0af6-48b2-b46d-e7a2ba12ad85)

one-community commented:
Thank you all, merging!
